native/src/main/java/io/ballerina/lib/confluent/avro/serdes/AvroSerializer.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.lib.confluent.avro.serdes;
+
+import io.ballerina.runtime.api.Environment;
+import io.ballerina.runtime.api.Future;
+import io.ballerina.runtime.api.PredefinedTypes;
+import io.ballerina.runtime.api.async.StrandMetadata;
+import io.ballerina.runtime.api.creators.TypeCreator;
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.types.UnionType;
+import io.ballerina.runtime.api.values.BArray;
+import io.ballerina.runtime.api.values.BObject;
+import io.ballerina.runtime.api.values.BTypedesc;
+
+import java.nio.ByteBuffer;
+
+import static io.ballerina.lib.confluent.avro.serdes.ModuleUtils.getModule;
+
+public class AvroSerializer {
+    private static final String FUNCTION = "deserializeData";
+
+    public static final StrandMetadata EXECUTION_STRAND = new StrandMetadata(
+            getModule().getOrg(),
+            getModule().getName(),
+            getModule().getMajorVersion(),
+            FUNCTION);
+
+    public static Object deserialize(Environment env, BObject kafkaSerDes, BArray data,
```

Comment:
It seems that the `AvroSerializer` class only contains a  `deserialize()` method. Is that intentional?

Reply:
Yes, here the requirement of the `deserialize()` method is to return the user-inferred type and we need to use a Java API for that